### PR TITLE
Merges changes from todo-mvp and fixes UI tests

### PR DESCRIPTION
**Pull request by JoseAlcerreca**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,8 +1,6 @@
-# Android Architecture Blueprints [beta] - MVP + Dagger2
+# Android Architecture Blueprints - MVP + Dagger2
 ### Summary
 
-Project contributors: [Saúl Molinero](https://github.com/saulmm)
```
Comment (digitalbuddha) [inline README.md]: want to add me here? :-)

File: todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/addedittask/AddEditTaskActivity.java [hunk 2]
```diff
@@ -36,16 +36,22 @@
...
     AddEditTaskFragment mFragment;
-    
+
     @Inject
     @Nullable
     String mTaskId;
 
+    // In a rotation it's important to know if we want to let the framework restore view state or
+    // need to load data from the repository. This is saved into the state bundle.
+    boolean mIsDataMissing = true;
```
Comment (digitalbuddha) [inline todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/addedittask/AddEditTaskActivity.java]: private?

File: todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/addedittask/AddEditTaskActivity.java [hunk 3]
```diff
@@ -66,6 +72,12 @@ protected void onCreate(Bundle savedInstanceState) {
             ActivityUtils.addFragmentToActivity(getSupportFragmentManager(),
                     addEditTaskFragment, R.id.contentFrame);
         }
+
+        // Prevent the presenter from loading data from the repository if this is a config change.
+        if (savedInstanceState != null) {
```
Comment (digitalbuddha) [inline todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/addedittask/AddEditTaskActivity.java]: nitpick, keep `onCreate` smaller by encapsulating into a `restoreState` method

File: todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/addedittask/AddEditTaskActivity.java [hunk 4]
```diff
@@ -78,10 +90,20 @@ private ActionBar setupToolbar() {
...
+        super.onSaveInstanceState(outState);
+    }
 
     @Override
     public boolean onSupportNavigateUp() {
         onBackPressed();
         return true;
     }
+
+    public boolean isDataMissing() {
```
Comment (digitalbuddha) [inline todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/addedittask/AddEditTaskActivity.java]: I think module is in same package, maybe change to package private?

File: todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/addedittask/AddEditTaskPresenter.java [hunk 5]
```diff
@@ -50,14 +52,29 @@
     @Nullable
     private String mTaskId;
 
+    // This is provided lazily because its value is determined in the Activity's onCreate. By
+    // calling it in takeView(), the value is guaranteed to be set.
+    private Lazy<Boolean> mIsDataMissingLazy;
```
Comment (digitalbuddha) [inline todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/addedittask/AddEditTaskPresenter.java]: not crazy about having the 2 variables.  Might make sense to instead create an injectable `TaskState` which contains the boolean, allowing mutation.  Can save for future pr :-)

File: todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/addedittask/AddEditTaskActivity.java (hunk 2 above)
Comment (JoseAlcerreca) [inline todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/addedittask/AddEditTaskActivity.java]: Why is my Studio not picking this? :S

File: todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/addedittask/AddEditTaskActivity.java (hunk 3 above)
Comment (JoseAlcerreca) [inline todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/addedittask/AddEditTaskActivity.java]: :+1: done

File: todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/addedittask/AddEditTaskActivity.java (hunk 4 above)
Comment (JoseAlcerreca) [inline todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/addedittask/AddEditTaskActivity.java]: Done

File: todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/addedittask/AddEditTaskPresenter.java (hunk 5 above)
Comment (JoseAlcerreca) [inline todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/addedittask/AddEditTaskPresenter.java]: Yeah I hate having the 2 variables. Let's do it in a future PR.

File: todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/addedittask/AddEditTaskPresenter.java [hunk 6]
```diff
@@ -80,7 +97,8 @@ public void populateTask() {
     @Override
     public void takeView(AddEditTaskContract.View view) {
         mAddTaskView = view;
-        if (!isNewTask()) {
+        mIsDataMissing = mIsDataMissingLazy.get();
+        if (!isNewTask() && mIsDataMissing) {
```
Comment (JoseAlcerreca) [inline todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/addedittask/AddEditTaskPresenter.java]: It's not really state is it? In any case, on a different PR.

**Pull request merged**